src/main/java/org/truffleruby/core/array/ArrayNodes.java

```diff
@@ -1581,6 +1581,62 @@ public DynamicObject replace(DynamicObject array, DynamicObject other,
 
     }
 
+    @Primitive(name = "array_rotate_inplace", needsSelf = false, lowerFixnum = 2)
+    @ImportStatic(ArrayGuards.class)
+    public abstract static class RotateInplaceNode extends PrimitiveArrayArgumentsNode {
+
+        @Specialization(guards = "strategy.matches(array)", limit = "ARRAY_STRATEGIES")
+        public DynamicObject rotate(DynamicObject array, int amount,
+                @Cached("of(array)") ArrayStrategy strategy) {
+            final int size = strategy.getSize(array);
+            assert 0 < amount && amount < size;
+            final ArrayMirror mirror = strategy.newMirror(array);
+
+            rotateReverse(amount, size, mirror);
```

Comment:
Maybe rename `size` to `arrayLength`. `amount` and `size` are very similar and it can be a bit confusing.

Reply:
I'm happy to rename `amount` to `n` or `rotation` or so, but I think `size` is consistent across the whole file so I'd rather not change it (arrayLength sounds to me like the length of the backing array).

Reply:
Fair enough.